web/src/components/Presentation/AttributeItem.vue

```diff
@@ -2,7 +2,63 @@
 import { defineComponent, PropType } from '@vue/composition-api';
 import { getField } from '@/encoding';
 import { fieldDisplayName, valueDisplayName } from '@/util';
-import { BaseSearchResult, BiosampleSearchResult } from '@/data/api';
+import { BaseSearchResult, BiosampleSearchResult, JSONValue } from '@/data/api';
+
+export enum Format {
+  RangeWithUnit = 'RangeWithUnit',
+  Range = 'Range',
+  Unit = 'Unit',
+}
+
+/**
+ * Helper function that returns a string representation of the depth in the specified format, if the specified
+ * depth annotation contains adequate information to do so. Otherwise, the function returns `undefined`.
+ */
+const buildStrFromDepthAnnotation = (depthAnnotation: JSONValue, format: Format): string | undefined => {
+  let str: string | undefined;
```

Comment:
This function is a little mysterious to me. My understanding is that it will generate a depth string for a given depth annotation according to the specified format. If the specified format is unsupported by the amount of data in the annotation, `undefined` is returned. But when `format === Format.Unit`, it only returns the unit, with no scalar value and the calling function is then responsible for picking the right scalar value for the unit.

Do we have a use case for getting the depth string using a format that doesn't include all possible information from the `Biosample` object? If not, then what if this function always returned the most detailed possible depth string?

If this function took the additional parameter of `topLevelDepth`, could it be used to generate the full string for every case?

I think I would have expected the function using this helper to do `depthString = buildStrFromDepthAnnotation(...)` just and have the string ready to go.

It might need to be renamed (since the current name only implies that it uses the annotation), but this change could clean up the changes in `AttributeList.vue`.

Reply:
Thanks for giving me detailed feedback about this function. The summary you wrote in the first paragraph is accurate.

> Do we have a use case for getting the depth string using a format that doesn't include all possible information from the `Biosample` object?

I don't know of such a use case.

> If not, then what if this function always returned the most detailed possible depth string?
>
> If this function... (the rest of your message)

I think that would be a more intuitive solution than the one I implemented. I'll refactor the code to do that.